### ISSUE TITLE
[Google3 Bug]: Android Talkback does not read "1 of 4" or similar indexes of items for any of the children of the `ListView` even with the use of `IndexedSemantics`

### ISSUE THREAD

**Issue by bhuvneshAgrawal** [flutter/flutter#168651]
### Help us understand the severity of this issue

- [ ] causing severe production issues e.g. malfunctions or data loss
- [ ] blocking next binary release
- [x] blocking a client feature launch within a quarter
- [ ] nice-to-have but does not block a launch within the next quarter

### Steps to reproduce

1. Run the flutter application on android device
2. Start android Talkback
3. Swipe horizontally to allow Android talkback read through the Text widgets on the screen.

### Expected results

With the use of IndexedSemantics, Android Talkback should have read:
"1 of 4, Title 1"
"2 of 4, Title 2",
"3 of 4, Title 3",
"4 of 4, Title 4"

### Actual results

The following is read out by Talkback:
"Title 1"
"Title 2"
"Title 3"
"Title 4"

### Code sample

<details open><summary>Code sample</summary>
In the `bug_repro` project (created via `flutter create ~/Desktop/bug_repro`),
Replace the `build` method of `_MyHomePageState` with the following code:

```dart
@override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: ListView(
          addSemanticIndexes: false,
          semanticChildCount: 4,
          physics: NeverScrollableScrollPhysics(),
          children: [
            IndexedSemantics(index: 0, child: Text('title 1')),
            IndexedSemantics(index: 1, child: Text('title 2')),
            IndexedSemantics(index: 2, child: Text('title 3')),
            IndexedSemantics(index: 3, child: Text('title 4')),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      )
    );
  }
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/0d3d1b93-8514-4db2-8f7e-7dc0b8f77516

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
$ flutter doctor -v
[✓] Flutter (Channel stable, 3.29.3, on macOS 15.4.1 24E263 darwin-arm64, locale en) [2.2s]
    • Flutter version 3.29.3 on channel stable at /Users/bhagrawal/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ea121f8859 (4 weeks ago), 2025-04-11 19:10:07 +0000
    • Engine revision cf56914b32
    • Dart version 3.7.2
    • DevTools version 2.42.3

[!] Android toolchain - develop for Android devices (Android SDK version 35.0.0) [2.3s]
    • Android SDK at /Users/bhagrawal/Library/Android/sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: /Applications/Android Studio with Blaze.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.6+-13368085-b895.109)
    ! Some Android licenses not accepted. To resolve this, run: flutter doctor --android-licenses

[!] Xcode - develop for iOS and macOS (Xcode 16.0) [1,204ms]
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 16A242d
    ✗ CocoaPods not installed.
        CocoaPods is a package manager for iOS or macOS platform code.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/to/platform-plugins
      For installation instructions, see https://guides.cocoapods.org/using/getting-started.html#installation

[✓] Chrome - develop for the web [8ms]
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2024.3) [7ms]
    • Android Studio at /Applications/Android Studio with Blaze.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.6+-13368085-b895.109)

[✓] VS Code (version 1.100.0) [6ms]
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.110.0

[✓] Connected device (4 available) [6.4s]
    • Pixel 6a (mobile)               • 29121JEGR04412        • android-arm64  • Android 15 (API 35)
    • macOS (desktop)                 • macos                 • darwin-arm64   • macOS 15.4.1 24E263
      darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad • darwin         • macOS 15.4.1 24E263
      darwin-arm64
    • Chrome (web)                    • chrome                • web-javascript • Google Chrome 136.0.7103.93
    ! Error: Browsing on the local area network for GBA iPhone12 202404. Ensure the device is unlocked and
      attached with a cable or associated with the same local area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code -27)
    ! Error: Browsing on the local area network for Bhuvnesh’s Apple Watch. Ensure the device is unlocked and
      discoverable via Bluetooth. (code -27)

[✓] Network resources [478ms]
    • All expected network resources are available.

! Doctor found issues in 2 categories.
```

</details>

b/418696524

**Comment by chunhtai** [comment]
native Android app also does not annunce "1 of 4" in a listview.

**Comment by bhuvneshAgrawal** [comment]
> [@bhuvneshAgrawal](https://github.com/bhuvneshAgrawal) Is there a criteria that require this behavior?

The VPAT A11Y evaluation and certification requires visual lists to be read out as lists of items with their indexes.

**Comment by chunhtai** [comment]
Talked with customer offline, in the IndexedSemantics doc promised to provide semantics index status in a ListView when using screen readers. However, it doesn't provide enough information to fullfil WCAG 1.3.1. As internal bug has mentioned, list litems  in a list view should pronounce  (item x of Y) simliar to TabBar.

We should make sure that IndexedSemantics provide these information.

**Comment by jwlilly** [comment]
@chunhtai, I did a search in the repo, but I couldn't find where these values were being set in the Android Accessibility API. But these are controlled using the AccessibilityNodeInfo.CollectionInfo and AccessibilityNodeInfo.CollectionItemInfo. 

- CollectionInfo: holds information related to the number of rows and columns. This is applied to the container element for the list items. It supports grids, but for lists, its just a single column and however many rows are needed. 
- CollectionItemInfo: holds information related to the list item position in the grid/list. This is applied to each list item. Information includes row index, number of rows spanned, column index, number of columns spanned

- [AccessibilityNodeInfo.CollectionInfo  |  API reference  |  Android Developers](https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo.CollectionInfo)
- [AccessibilityNodeInfo.CollectionItemInfo  |  API reference  |  Android Developers](https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo.CollectionItemInfo)

**Comment by chunhtai** [comment]
We did set the collection info https://github.com/flutter/flutter/blob/268a5e387c7fe8aebde6e4db95c737d08626dc51/engine/src/flutter/shell/platform/android/io/flutter/view/AccessibilityBridge.java#L861

but we didn't set the CollectionItemInfo, maybe that is the cause? will need to verify

**Comment by jwlilly** [comment]
I guess I missed that in my search! But a missing CollectionItemInfo sounds like the issue. With only the CollectionInfo, it will announce only the total number of items but not the index of each item

**Comment by chunhtai** [comment]
still trying to find an owner for this issue. no estimate fix day yet

**Comment by jwlilly** [comment]
I'd like to help out if that's possible. I tried compiling the engine several months ago but ran into environment issues. If there's someone that can help me get set up or a step by step tutorial, that would be helpful.

**Comment by chunhtai** [comment]
If you have engine compile issue you can ask in discord hackers channel, or feel free to ping me in hackers-accessibility channel, my discord name is the same as my github id

**Comment by chunhtai** [comment]
@jwlilly are you still working on this issue?

No pressure, asking just for book keeping purpose.

**Comment by jwlilly** [comment]
> [@jwlilly](https://github.com/jwlilly) are you still working on this issue?
> 
> No pressure, asking just for book keeping purpose.

Yep. I got sidetracked. I need to sign the commit and start the PR process

**Comment by victorsanni** [comment]
Reopening due to revert.
